packages/cdktf-cli/src/lib/dependencies/dependency-manager.ts

```diff
@@ -137,12 +137,13 @@ export class DependencyManager {
     );
   }
 
-  async addProvider(
-    constraint: ProviderConstraint
-  ): Promise<{ addedLocalProvider: boolean }> {
+  async addProvider(constraint: ProviderConstraint): Promise<{
+    addedLocalProvider: boolean;
+    prebuiltProviderToAdd?: PackageConstraint;
```

Comment:
maybe name this `prebuiltProviderToInstall` to denote that this is a dependency in the target language's depedency manager? "Add" seems a bit overused in this context.